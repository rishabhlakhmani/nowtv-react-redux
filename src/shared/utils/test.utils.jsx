import React from 'react'
import { render } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import membersReducer from '../../store/members/members.slice'
import messagesReducer from '../../store/members/members.slice'

export function renderWithProviders(
  ui,
  {
    preloadedState = {
        messages: {
            messages: []
        },
        members: {
            members: []
        },
    },
    store = configureStore({
      reducer: { messages: messagesReducer, members: membersReducer },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}
