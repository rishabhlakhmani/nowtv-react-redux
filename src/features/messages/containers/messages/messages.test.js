import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Messages from './messages.container';
import { renderWithProviders } from '../../../../shared/utils/test.utils';

describe('Messages', function () {
  it('should render the component', function () {
    let messages = renderWithProviders(
      <MemoryRouter>
        <Messages />
      </MemoryRouter>
    );
    expect(messages.container).toMatchSnapshot();
  });

  it('should render the Messages List', function () {
    const initialMessages = [
      {
        id: 'fa94a88c-35bb-4a22-baa8-19910a883a6b',
        userId: '25bc59fd-1832-4eb4-964b-ff319e763494',
        message: 'Nunc rhoncus dui vel sem.',
        timestamp: '2016-09-09T05:04:05Z',
      },
    ];

    const { getByText } = renderWithProviders(
      <MemoryRouter>
        <Messages />
      </MemoryRouter>,
      {
        preloadedState: {
          messages: {
            messages: initialMessages,
          },
        },
      }
    );

    expect(getByText(initialMessages[0].message)).toBeInTheDocument();
    expect(getByText(initialMessages[0].message)).toBeInTheDocument();
  });

  it('should render No Messages available. when there is no message', function () {
    const { getByText } = renderWithProviders(
      <MemoryRouter>
        <Messages />
      </MemoryRouter>
    );

    expect(getByText(/No Messages available/i)).toBeInTheDocument();
  });
});
