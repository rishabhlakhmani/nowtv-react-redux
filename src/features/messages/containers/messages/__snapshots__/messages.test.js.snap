// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should render the component 1`] = `
<div>
  <div
    class="mx-auto d-flex justify-content-center flex-wrap page-container container"
  >
    <header
      class=""
    >
      <h2>
        Messages List
      </h2>
    </header>
    <div
      class="w-100"
    >
      <main>
        <div>
           No Messages available. 
        </div>
      </main>
    </div>
  </div>
</div>
`;
